tests/unit/test_client.py

```diff
@@ -1607,6 +1607,25 @@ def test_client_close_context_manager(self):
 
         self.endpoint.close.assert_called_once_with()
 
+    def test_sslCommonName_warning(self):
+        creator = self.create_client_creator()
+        self.endpoint_data['sslCommonName'] = 'bar'
+        with self.assertWarns(UserWarning) as warning:
+            creator.create_client(
+                'myservice', 'us-west-2', credentials=self.credentials
+            )
+            self.assertEqual(len(warning.warnings), 1)
+            warning_message = warning.warnings[0].message
+            ssl = self.endpoint_data['sslCommonName']
+            host = self.endpoint_data['hostname']
+            expected_warning_message = (
+                "The myservice client is currently using a deprecated endpoint:"
+                f" {ssl}. In the next minor version this will be moved to "
+                f"{host}. See https://github.com/boto/botocore/issues/2376 for "
+                "more details."
+            )
+            self.assertEqual(str(warning_message), expected_warning_message)
+
```

Comment:
Let's avoid hardcoding the warning message here so we're not having to update the test every time it changes. One additional test we can do though is a second client creation and asserting it doesn't create a second warning.

```python
with pytest.warns(None) as warning:
    creator.create_client(
        'myservice', 'us-west-2', credentials=self.credentials
    )
self.assertEqual(len(warning.warnings), 0)
```

Reply:
per our offline conversation `pytest.warns` doesn't work as expected and seemingly warnings are still thrown under the hood, just suppressed/not printed to the console. so I didn't add a new test